watertap/tools/parameter_sweep.py

```diff
@@ -319,19 +342,460 @@ def _interp_nan_values(global_values, global_results):
+                        fname="output_dict.h5"):
+
+    fpath = os.path.join(output_directory, fname)
+    f = h5py.File(fpath, 'w')
+    for key, item in output_dict.items():
+        grp = f.create_group(key)
+        if key != "solve_status":
+            for subkey, subitem in item.items():
+                subgrp = grp.create_group(subkey)
+                for subsubkey, subsubitem in subitem.items():
+                    if subsubkey == 'lower bound' and subsubitem is None:
+                        subgrp.create_dataset(subsubkey, data=np.finfo('d').min)
+                    elif subsubkey == 'upper bound' and subsubitem is None:
+                        subgrp.create_dataset(subsubkey, data=np.finfo('d').max)
+                    else:
+                        subgrp.create_dataset(subsubkey, data=output_dict[key][subkey][subsubkey])
+        elif key == 'solve_status':
+            grp.create_dataset(key, data=output_dict[key])
+        else:
+            pass
+
+    f.close()
+
+# ================================================================
+
+def _read_output_h5(filepath):
+
+    f = h5py.File(filepath , 'r')
+
+    l1_keys = list(f.keys())
+    output_dict = {}
+    for key in l1_keys: # Input or Output
+        if key != 'solve_status':
+            output_dict[key] = {}
+            l2_keys = list(f[key].keys())
+            for subkey in l2_keys: # Variable name
+                output_dict[key][subkey] = {}
+                l3_keys = list(f[key][subkey].keys())
+                for subsubkey in l3_keys: # variable metadata
+                    output_dict[key][subkey][subsubkey] = f[key][subkey][subsubkey][()]
+                    if subsubkey == "units":
+                        # The strings are recovered in bytes. we choose to convert it to utf-8
+                        output_dict[key][subkey][subsubkey] = output_dict[key][subkey][subsubkey].decode("utf-8")
+        elif key == 'solve_status':
+            output_dict[key] = [x.decode("utf-8") for x in list(f[key]['solve_status'][()])]
+            output_dict[key] = output_dict[key]
+
+    f.close()
+
+    return output_dict
+
+# ================================================================
+
+def _numeric_termination_condition_translation(termination_condition):
+
+    pyomo_termination_condition = TerminationConditionMapping()
+
+    ctr = 0
+    if isinstance(termination_condition, str):
+        if termination_condition in pyomo_termination_condition.str_to_int_mapping:
+            ctr += 1
+            return pyomo_termination_condition.str_to_int_mapping[termination_condition]
+    elif isinstance(termination_condition, np.integer):
+        if termination_condition in pyomo_termination_condition.int_to_str_mapping:
+            ctr += 1
+            return pyomo_termination_condition.int_to_str_mapping[termination_condition]
+
+    if ctr == 0:
+        raise ValueError("Pyomo termination condition value not found in known list")
```

Comment:
I don't really understand why we wouldn't know which direction (from int to TerminationCondition or TerminationCondition to int).

If we know which direction we're going, we should just access the dictionary directly, as opposed to calling this function.

Reply:
Yeah, I agree. This can go away. I'll make the change.

Reply:
This function was removed in commit cc8ee3ed823f1e41fbf99499f33672e1c17ff20a